src/wallet/wallet.h

```diff
@@ -1661,16 +1674,16 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     bool SaveRecipientMappings(const uint256& txid, const std::vector<RecipientMapping>& recipients)
     {
         LOCK2(cs_main, cs_wallet);
-        LogPrintf("SaveRecipientMappings:\n%s", txid.ToString());
 
         for (const auto& recipient : recipients)
         {
+            sendRecipients[txid].push_back(recipient);
             if (recipient.ua.has_value()) {
-                CWalletDB(strWalletFile).WriteRecipientMapping(
+                assert(CWalletDB(strWalletFile).WriteRecipientMapping(
```

Comment:
Should this be (logging and) propagating the error rather than asserting?

Reply:
Do we consider a database write failure to be a recoverable error? The way that we use `assert` in general is mixed up, as we've discussed in the past, but I think that write failures should generally be catastrophic.

Reply:
Fixed in #5715